src/App.js

```diff
@@ -1,24 +1,89 @@
-import logo from './logo.svg';
-import './App.css';
+import React, { useState } from "react";
+import { BsPlusCircleFill } from "react-icons/bs";
+import { MdDelete } from "react-icons/md";
 
 function App() {
+  const [todos, setTodos] = useState(
+    () => JSON.parse(localStorage.getItem("todos")) || []
+  );
+
+  const [todo, setTodo] = useState("");
+  const [error, setError] = useState(false);
+
+  const handleChange = (e) => {
+    // This function is called when the user types in the input field
+    setError(false);
+    setTodo(e.target.value);
+  };
+  const deleteTask = (index) => {
+    // This function is called when the user clicks on the delete icon to delete a todo item
+    const newTodos = [...todos];
+    newTodos.splice(index, 1);
+    setTodos(newTodos);
+    localStorage.setItem("todos", JSON.stringify(todos));
+  };
+  const handleCheck = (index) => {
+    // This function is called when the user clicks on the checkbox to mark a todo item as completed
+    const newTodos = [...todos];
+    newTodos[index].completed = !newTodos[index].completed;
+    setTodos(newTodos);
+    localStorage.setItem("todos", JSON.stringify(newTodos));
+  };
+  const handleClick = (e) => {
+    //This function is called when the user clicks on the plus icon to add a new todo item
+    if (todo.trim().length <= 0) {
+      setError(true);
+    } else {
+      const newTodo = [{ task: todo, completed: false }, ...todos];
+      setTodos(newTodo);
+      localStorage.setItem("todos", JSON.stringify(newTodo));
+    }
+  };
   return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
+    <main className="max-w-5xl mx-auto">
+      <h1 className="text-7xl max font-bold text-current-900 text-center">
+        todos
+      </h1>
+      <div
+        className={`flex items-center justify-between shadow-3xl m-4 rounded-full px-4 ${
+          error && "border-red-500 border"
+        }`}
+      >
+        <input
+          onChange={handleChange}
+          value={todo}
+          className="px-4 py-2 focus:outline-none flex-1"
+          placeholder="add todo..."
+        />
+
+        <BsPlusCircleFill
+          onClick={() => handleClick()}
```

Comment:
```suggestion
          onClick={handleClick}
```

can also work because the handleClick does not receive any parameter